oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/LastRevRecoveryAgent.java

```diff
@@ -142,9 +143,29 @@ public int recover(int clusterId, long waitUntil)
         ClusterNodeInfoDocument nodeInfo = missingLastRevUtil.getClusterNodeInfo(clusterId);
 
         if (nodeInfo != null) {
+            // Check our own lease before running recovery for another
+            // clusterId (OAK-9656)
+            long now = revisionContext.getClock().getTime();
+            ClusterNodeInfoDocument me = null;
+            if (clusterId != revisionContext.getClusterId()) {
+                // Get leaseEnd from our own cluster node info, unless
+                // we are doing recovery on startup for the clusterId
+                // we want to acquire. Then it's fine to go ahead with
+                // an expired lease.
+                me = missingLastRevUtil.getClusterNodeInfo(revisionContext.getClusterId());
+            }
+            if (me != null && me.isRecoveryNeeded(now)) {
```

Comment:
Since `me` is only used within the above 
```if (clusterId != revisionContext.getClusterId()) {```
block, this block here could be moved 1 line up into that block. That could make it slightly easier to read - since the scope it then smaller. wdyt?

Reply:
Good suggestion. Will do.